h2o-jetty-8/src/main/java/water/webserver/jetty8/Jetty8Helper.java

```diff
@@ -0,0 +1,199 @@
+import water.webserver.iface.H2OHttpView;
+import water.webserver.iface.LoginType;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+
+class Jetty8Helper {
+
+  private final H2OHttpConfig config;
+  private final H2OHttpView h2oHttpView;
+
+  Jetty8Helper(H2OHttpView h2oHttpView) {
+    this.h2oHttpView = h2oHttpView;
+    this.config = h2oHttpView.getConfig();
+  }
+
+  Server createJettyServer(String ip, int port) {
+    System.setProperty("org.eclipse.jetty.server.Request.maxFormContentSize", Integer.toString(Integer.MAX_VALUE));
+
+    final Server jettyServer = new Server();
+    jettyServer.setSendServerVersion(false);
+
+    final Connector connector;
+    final String proto;
+    if (config.jks != null) {
+      proto = "https";
+      final SslContextFactory sslContextFactory = new SslContextFactory(config.jks);
+      sslContextFactory.setKeyStorePassword(config.jks_pass);
+      connector = new SslSocketConnector(sslContextFactory);
+    } else {
+      proto = "http";
+      connector = new SocketConnector();
+    }
+    if (ip != null) {
+      connector.setHost(ip);
+    }
+    connector.setPort(port);
+    configureConnector(proto, connector);
+    jettyServer.setConnectors(new Connector[]{connector});
+    return jettyServer;
+  }
+
+  // Configure connector via properties which we can modify.
+  // Also increase request header size and buffer size from default values
+  // located in org.eclipse.jetty.http.HttpBuffersImpl
+  // see PUBDEV-5939 for details
+  private void configureConnector(String proto, Connector connector) {
+    connector.setRequestHeaderSize(getSysPropInt(proto+".requestHeaderSize", 32*1024));
+    connector.setRequestBufferSize(getSysPropInt(proto+".requestBufferSize", 32*1024));
+    connector.setResponseHeaderSize(getSysPropInt(proto+".responseHeaderSize", connector.getResponseHeaderSize()));
+    connector.setResponseBufferSize(getSysPropInt(proto+".responseBufferSize", connector.getResponseBufferSize()));
+  }
+
+  private static int getSysPropInt(String suffix, int defaultValue) {
+    return Integer.getInteger(H2OHttpConfig.SYSTEM_PROP_PREFIX + suffix, defaultValue);
+  }
+
+  HandlerWrapper authWrapper(Server jettyServer) {
+    if (config.loginType == LoginType.NONE) {
+      return jettyServer;
+    }
+
+    // REFER TO http://www.eclipse.org/jetty/documentation/9.1.4.v20140401/embedded-examples.html#embedded-secured-hello-handler
+    final LoginService loginService;
+    switch (config.loginType) {
+      case HASH:
+        loginService = new HashLoginService("H2O", config.login_conf);
+        break;
+      case LDAP:
+      case KERBEROS:
+      case PAM:
+        loginService = new JAASLoginService(config.loginType.jaasRealm);
+        break;
+      default:
+        throw new UnsupportedOperationException(config.loginType + ""); // this can never happen
```

Comment:
It can, when somebody adds new `LoginType` and forgets about this part. In an ideal world, there is a test for that. Up to you. I don't think it is completely necessary.